packages/sw-precaching/src/lib/models/precache-entries/object-precache-entry.js

```diff
@@ -1,43 +1,43 @@
-import ErrorFactory from '../error-factory';
-import RevisionedCacheEntry from './revisioned-cache-entry.js';
+import ErrorFactory from '../../error-factory';
+import RevisionedCacheEntry from './base-precache-entry.js';
 
 class DefaultsCacheEntry extends RevisionedCacheEntry {
-  constructor({revisionID, revision, url, cacheBust}) {
+  constructor({entryID, revision, url, cacheBust}) {
     // Check cacheBust is a boolean
     let filteredCacheBust = cacheBust;
     if (typeof filteredCacheBust === 'undefined') {
       filteredCacheBust = true;
     } else if (typeof filteredCacheBust !== 'boolean') {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad cacheBust Parameter. It should be a boolean value: ' +
           JSON.stringify(cacheBust)));
     }
 
     // Check revision is a string with a length
     if (typeof revision !== 'string' || revision.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad revision Parameter. It should be a string with at ' +
           'least one character: ' + JSON.stringify(revision)));
     }
 
     // Check url is a string.
     if (typeof url !== 'string' || url.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad url Parameter. It should be a string:' +
           JSON.stringify(url)));
     }
 
-    // Check revisionID type
-    if (typeof revisionID === 'undefined') {
-      revisionID = new URL(url, location.origin).toString();
-    } else if (typeof revisionID !== 'string' || revisionID.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
-        new Error('Bad revisionID Parameter. It should be a string with at ' +
-          'least one character: ' + JSON.stringify(revisionID)));
+    // Check entryID type
+    if (typeof entryID === 'undefined') {
+      entryID = new URL(url, location.origin).toString();
```

Comment:
This shouldn't be`location.origin`, which would cause all relative URLs to be resolved with `/` as the base.

Using `location` would match what `sw-precache` [currently does](https://github.com/GoogleChrome/sw-precache/blob/master/service-worker.tmpl#L56), i.e. use the SW's script location as the base.

There's an argument to be made that using `registration.scope` as the base would make more sense than using the SW script location. I could see that as being more intuitive. (`self.registration.scope` wasn't implemented when `sw-precache` first came out, so it's a historical accident that it's not used.) But it could make it harder to transition from using `sw-precache` to using the new library for developers who also set custom scopes.

Thoughts?